### DISCUSSION TITLE
Kubernetes file storage

### DISCUSSION THREAD

**Discussion by sfusco-kryptowire** [micromdm/micromdm#863] [Q&A] [answered]
I'm looking to deploy micromdm inside a kubernetes cluster.  My understanding is that it does not use postgres or any kind of RDBMS, but instead uses files with BoltDB?  What kind of persistent storage do I need to consider giving it and to which file or directory paths so that things aren't blown away every time it restarts?

**Comment by korylprince** [comment, marked as answer]
You pass `-config-path` to micromdm serve to set a directory for the database. The BoltDB database is a single file placed at `$(config-path)/micromdm.db`. You can treat it much like you would treat a SQLite database. MicroMDM needs read/write access and (AFAIK) maintains an open file handle on it. I don't have a ton of experience with Kubernetes, but I'd expect you'd need a PersistentVolumeClaim for the database.

The only other thing stored in the config directory is certificates if you're using MicroMDM's autocert functionality (e.g. Let's Encrypt). I'm guessing that since you're using Kubernetes, you're probably taking care of that with a proxy.

Another option you might consider is using [nanoMDM](https://github.com/micromdm/nanomdm), which has more traditional storage options (MySQL, Postgres, etc), but less built in features than MicroMDM.
